packages/opentelemetry-core/src/common/attributes.ts

```diff
@@ -16,24 +16,26 @@
 
 import { diag, SpanAttributeValue, SpanAttributes } from '@opentelemetry/api';
 
-export function sanitizeAttributes(attributes: unknown): SpanAttributes {
-  const out: SpanAttributes = {};
-
+export function sanitizeAttributes(
+  attributes: unknown,
+  out: SpanAttributes = {}
+): SpanAttributes {
   if (typeof attributes !== 'object' || attributes == null) {
     return out;
   }
 
-  for (const [key, val] of Object.entries(attributes)) {
+  for (const key in attributes) {
```

Comment:
There are some environments (runtimes) where the usage of `x in y` can cause issues where the only additional fix is to also use `Object.hasOwnProperty` so this may have unexpected (endless looping) consequences.

It's been a while since I saw this previously, but (from memory) it is caused by a 3rd party library adding stuff to Object to support the runtime environment.

Reply:
I propose to add `Object.hasOwnProperty` then, I think allocations is more of a problem than sheer performance

Reply:
Added propertyIsEnumerable check